pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -286,6 +291,11 @@ func (g *DefaultValidator) validateTrialTemplate(instance *experimentsv1beta1.Ex
 		trialParametersNames[parameter.Name] = true
 		trialParametersRefs[parameter.Reference] = true
 
+		// Check if parameter reference exist in experiment parameters
+		if _, ok := experimentParameterNames[parameter.Reference]; !ok {
+			return fmt.Errorf("parameter reference %v does not exist in spec.parameters: %v", parameter.Reference, instance.Spec.Parameters)
+		}
```

Comment:
I think for NAS Experiments that doesn't work.
For example, for the [ENAS Experiment](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/nas/enas-cpu.yaml#L126-L132), the Suggestion returns `architecture` and `nn_config` which don't exist in `spec.Parameters`.
For HP Experiments, currently all parameters that Suggestions return exist in `spec.parameters`, but I am not sure if for other AutoML techniques that always be true (for example for [PBT](https://github.com/kubeflow/katib/issues/1382), AutoMC, etc.).

I think for now, we can check if  `experimentParameterNames` is not empty than we can apply this check.

WDYT @henrysecond1 @anencore94 @gaocegege @johnugeorge ?

Reply:
SGTM!

Reply:
@andreyvelich Thank you for pointing it out. Sounds good to me, too :)

Reply:
Yes that make sense ! Thanks